### PR TITLE
[KOGITO-1920] - Example for persistence in Mongo DB with Quarkus(native image )

### PR DESCRIPTION
**Pull request by swati-kale**
Signed-off-by: Swati Kale <swkale@redhat.com>

Many thanks for submitting your Pull Request :heart:! 

Please make sure that your PR meets the following requirements:

- [ ] You have read the [contributors guide](CONTRIBUTING.md)
- [ ] Pull Request title is properly formatted: `KOGITO-XYZ Subject`
- [ ] Pull Request title contains the target branch if not targeting master: `[0.9.x] KOGITO-XYZ Subject`
- [ ] Pull Request contains link to the JIRA issue
- [ ] Pull Request contains link to any dependent or related Pull Request
- [ ] Pull Request contains description of the issue
- [ ] Pull Request does not include fixes for issues other than the main ticket

### CODE REVIEW

File: kogito-quarkus-extension/deployment/src/main/java/org/kie/kogito/quarkus/deployment/KogitoAssetsProcessor.java [hunk 1]
```diff
@@ -287,6 +288,18 @@ private void generatePersistenceInfo(AppPaths appPaths, IndexView inputIndex) th
         if (usePersistence) {
             resource.produce(new NativeImageResourceBuildItem("kogito-types.proto"));
         }
+        
+        if(!parameters.isEmpty() && parameters.contains("com.mongodb.client.MongoClient")) {
```
Comment (cristianonicolai) [inline kogito-quarkus-extension/deployment/src/main/java/org/kie/kogito/quarkus/deployment/KogitoAssetsProcessor.java:292]: @swati-kale I think the solution is good but we shouldnt depend on the mongo client parameter to define if the native items should be produced. I think a better approach would be to check the persistence type. Perhaps extract the logic from the PersistenceGenerator and reuse the type here and as parameter to the generator.

**Comment by cristianonicolai** [comment]
ok to test

File: kogito-quarkus-extension/deployment/src/main/java/org/kie/kogito/quarkus/deployment/KogitoAssetsProcessor.java (hunk 1 above)
Comment (swati-kale) [reply kogito-quarkus-extension/deployment/src/main/java/org/kie/kogito/quarkus/deployment/KogitoAssetsProcessor.java]: ok, done,

File: kogito-codegen/src/main/java/org/kie/kogito/codegen/process/persistence/PersistenceGenerator.java [hunk 2]
```diff
@@ -66,8 +66,8 @@
 
     private static final String FILESYSTEM_PERSISTENCE_TYPE = "filesystem";
     private static final String INFINISPAN_PERSISTENCE_TYPE = "infinispan";
```
Comment (cristianonicolai) [inline kogito-codegen/src/main/java/org/kie/kogito/codegen/process/persistence/PersistenceGenerator.java:68]: just to make it consistent with the other tyes
```suggestion
    public static final String FILESYSTEM_PERSISTENCE_TYPE = "filesystem";
    public static final String INFINISPAN_PERSISTENCE_TYPE = "infinispan";
```

**Review by MarianMacik — APPROVED** [review]

**Review by cristianonicolai — APPROVED** [review]

**Pull request merged**